src/parser/demonhunter/havoc/modules/spells/azeritetraits/FuriousGaze.js

```diff
@@ -47,11 +52,21 @@ class FuriousGaze extends Analyzer{
     this.statTracker.add(SPELLS.FURIOUS_GAZE_BUFF.id, {
       haste,
     });
-    this.addEventListener(Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.FURIOUS_GAZE_BUFF), this.onBuffEvent);
+    this.addEventListener(Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.FURIOUS_GAZE_BUFF), this.onFuriousGazeApply);
+    this.addEventListener(Events.removebuff.by(SELECTED_PLAYER).spell(SPELLS.FURIOUS_GAZE_BUFF), this.onFuriousGazeRemove);
   }
 
-  onBuffEvent(event) {
-    this.furiousGazeProcsCounter += 1;
+  onFuriousGazeApply(event) {
+    if (this.furiousGazeStartTime === undefined) {
+      this.furiousGazeStartTime = event.timestamp;
+    }
+  }
+
+  onFuriousGazeRemove(event) {
+    if (this.furiousGazeStartTime !== undefined) {
+      this.furiousGazeProcsCounter += Math.round((event.timestamp - this.furiousGazeStartTime) / FURIOUS_GAZE_DURATION);
```

Comment:
I don't really understand what is going on. Doesn't the buff not get applied at all if you didn't finish the cast? In that case why does the player get partial credit for the cast?

Reply:
If the player currently has the buff and manages to get it a second time within the first buff's duration (Going Eyebeam, Meta reset through Chaotic Transformation, Eyebeam) there is no second applybuff event. I have added more decomposition and explanation in a new commit to clarify.

Reply:
And the rounding ist just because some events are 11995 or 12012 instead of 12000 (But im assuming you weren't pointing at that?)